cli/internal/cmd/verify.go

```diff
@@ -447,6 +463,156 @@ func (f *attestationDocFormatterImpl) parseSNPReport(b *strings.Builder, reportB
 	return nil
 }
 
+func parseMAAToken(ctx context.Context, b *strings.Builder, rawToken, attestationServiceURL string) error {
+	var claims maaTokenClaims
+	_, err := jwt.ParseWithClaims(rawToken, &claims, keyFromJKUFunc(ctx, attestationServiceURL), jwt.WithIssuedAt())
+	if err != nil {
+		return err
+	}
+
+	out, err := json.MarshalIndent(claims, "\t\t", "  ")
+	if err != nil {
+		return err
+	}
+
+	b.WriteString("\tMicrosoft Azure Attestation Token:\n\t")
+	b.WriteString(string(out))
+	return nil
+}
+
+// keyFromJKUFunc returns a function that gets the JSON Web Key URI from the token
+// and fetches the key from that URI. The keys are then parsed, and the key with
+// the kid that matches the token header is returned.
+func keyFromJKUFunc(ctx context.Context, webKeysURL string) func(token *jwt.Token) (any, error) {
+	return func(token *jwt.Token) (any, error) {
+		if token.Header["alg"] != "RS256" {
+			return nil, fmt.Errorf("invalid signing algorithm: %s", token.Header["alg"])
+		}
+		kid, ok := token.Header["kid"].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid kid: %v", token.Header["kid"])
+		}
+		jku, ok := token.Header["jku"].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid jku: %v", token.Header["jku"])
+		}
+		if jku != webKeysURL+"/certs" {
+			return nil, fmt.Errorf("jku from token (%s) does not match configured attestation service (%s)", jku, webKeysURL)
+		}
+
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		keySetBytes, err := httpGet(ctx, jku)
+		if err != nil {
+			return nil, fmt.Errorf("getting signing keys from jku %s: %w", jku, err)
+		}
+
+		var rawKeySet struct {
+			Keys []struct {
+				X5c []string
+				Kid string
+			}
+		}
+
+		if err := json.Unmarshal(keySetBytes, &rawKeySet); err != nil {
+			return nil, err
+		}
+
+		for _, key := range rawKeySet.Keys {
+			if key.Kid != kid {
+				continue
+			}
+			rawCert, _ := base64.StdEncoding.DecodeString(key.X5c[0])
+			cert, err := x509.ParseCertificate(rawCert)
```

Comment:
```suggestion
				X5c [][]byte
				Kid string
			}
		}

		if err := json.Unmarshal(keySetBytes, &rawKeySet); err != nil {
			return nil, err
		}

		for _, key := range rawKeySet.Keys {
			if key.Kid != kid {
				continue
			}
			cert, err := x509.ParseCertificate(key.X5c[0])
```
I think you can slightly simplify it like this

Reply:
No, this doesn't work as the cert is base64 and ParseCertificate expects DER.

Reply:
If you change the type of `X5c` to `[][]byte`, the base64 is decoded on unmarshaling.
I've done it like this here:
https://github.com/edgelesssys/go-azguestattestation/blob/a683be600fcfb6cca3e7c1e65216657a3a4d8c92/maa/validatetoken.go#L67-L106

But I'm also fine with keeping it as is.

Reply:
I've tried and it failed due to bad input. *confused*